src/Microsoft.DotNet.Darc/src/Darc/Operations/AddBuildToChannelOperation.cs

```diff
@@ -262,6 +262,11 @@ private async Task<int> PromoteBuildAsync(Build build, List<Channel> targetChann
                 { "ArtifactsPublishingAdditionalParameters", _options.ArtifactPublishingAdditionalParameters }
             };
 
+            if (build.GitHubBranch.Contains("release/", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
This will work but you should also compare against the AzDO branch AzureDevOpsBranch for cases where the branch is internal only.

Reply:
now checks both and also accommodates null for either, just in case I missed some corner case.